### ISSUE TITLE
Seasonal Search (relative date)

### ISSUE THREAD

**Issue by tomtjes** [TandoorRecipes/recipes#2357]
### Is your feature request related to a problem? Please describe.

I often find myself looking for recipes that I made a year ago or two years ago. This is a way for me to filter for recipes with ingredients that are currently in season. This is not easily doable with currently available search filters.

### Describe the solution you'd like

I'm proposing a search filter that allows to search for recipes `cooked 1 year ago`. Perhaps other people have use for other relative date searches like `3 months ago ` as well.

An advanced version would be a search option like `cooked around this time in previous years`. It would return all recipes sorted by time difference to the current day and month, disregarding the year. E.g.:

``` 
Today's date: Mar 5 (2023)

Search result for recipes made around this time of year:

- Mushroom Risotto (made Mar 5 2020)
- Butternut Squash Soup (made Mar 6 2019)
- Baked Potatoes (made Mar 4 2021)
```

There should probably be some logic that ensures that only the cook log entry that is closest to the current day and month is considered, so that recipes with multiple cook log entries only appear once in the list.

### Describe alternatives you've considered

Current search options include a filter for `last cooked` with the option to search before *or* after a given date. If it were possible to add this filter multiple times, a user could define one or multiple time ranges, e.g. `(before Apr 5 2022 AND after Feb 5 2022) OR (before Apr 5 2021 AND after Feb 5 2021)`

**Comment by mkieledunsche** [comment]
Hi @tomtjes
Just had a look at the search in Tandoor2 and saw adding multiple filters is possible now (see screenshot below).


your-tandoor-instance/advanced-search?cookedonGte=2025-06-01&createdonLte=2025-06-15
![Image](https://github.com/user-attachments/assets/c8379eef-efc0-41a7-8bd7-e46c48753998)

**Comment by tomtjes** [comment]
This results in all recipes cooked before 15.6.2025 because there is an imaginary OR between the fields.

**Comment by tomtjes** [comment]
I think this could work well as one of the rows of suggested recipes on the new start page. "Seasonal Favorites" could feature the highest rated recipes from a 12(?)-week window around today's date in previous years.

**Comment by tomtjes** [comment]
@vabene1111 Would you be open to a PR for this?

- add a search filter for seasonality: date picker returning recipes previously cooked around that time of year
- add a scroll row on the homepage with seasonal recipes

**Comment by smilerz** [comment]
You can search for items cooked between dates and save it as a Book

**Comment by tomtjes** [comment]
Yes, but what I'm looking for is, for instance: "What have I cooked around Sep 12 in 2024, 2023, 2022 (and all previous years)?" I think that would be a good way to find Christmas/summer/early spring etc. favorites. Plus, the row on the homepage would automatically update every day, providing inspiration from previous years without having to make a bunch of search filters.

**Comment by vabene1111** [comment]
To be honest I am not entirely sure about this. The search filters are incredibly complex and I am very careful adding new ones. That said I do not really see any other/better way to implement them. 

Regarding your general solution: This would only work if you use the log cooking function. If you do not use it you dont have a way to set a season. You could just use keywords but that requires manual effort. Also what about meal plans, should they be considered "cooked last year around this time". 

Another idea would be to assign a season to the foods (something I wanted to do for a long time) and search recipes based on that but that again has performance issues.

My favourite would be to rework the search system before adding anything new but I know that I won't be able to work on that for a while given all the other priorities. 

just some thoughts, interested in your feedback

**Comment by tomtjes** [comment]
Yeah, I saw that the filters require some effort... 

I am currently tagging my recipes with season keywords, but it's never perfect. Having a season property for foods would work for some things, but not others, e.g. recipes that use ingredients that are available year-round, but are still mostly made during a certain time of year, like German Zwiebelkuchen or American Apple Cider Donuts.

My suggestion is that I'll see if I can make a proof-of-concept based on cook logs. I think it should then be easy to swap-in/add meal plan entries. Then you can still decide if it's a good idea or not, no offense taken if it doesn't make the cut.

**Comment by smilerz** [comment]
A suggestion to make this a little more reusable beyond this niche case: 
- expand search by dates to allow exclusion of year 
- expand search to allow searching for recipes included in meal plans by various attributes (including between dates and between dates without years)

**Comment by tomtjes** [comment]
@smilerz these are suggestions for additional search filters, right?

**Comment by smilerz** [comment]
correct

**Comment by vabene1111** [comment]
Feel free to paly around with it, the search filters are already a big mess so I dont really care if we add even more mess to it. At some point I would like to clean them up tough, if anyone wants to volunteer doing that let me know :)
